handler/src/test/java/io/netty/handler/ssl/ocsp/OcspTest.java

```diff
@@ -0,0 +1,464 @@
+                }
+            } finally {
+                ReferenceCountUtil.release(serverSslContext);
+            }
+        } finally {
+            ssc.delete();
+        }
+    }
+
+    private static Channel newServer(EventLoopGroup group, SocketAddress address,
+            SslContext context, byte[] response, ChannelHandler handler) {
+
+        ServerBootstrap bootstrap = new ServerBootstrap()
+                .channel(LocalServerChannel.class)
+                .group(group)
+                .childHandler(newServerHandler(context, response, handler));
+
+        return bootstrap.bind(address)
+                .syncUninterruptibly()
+                .channel();
+    }
+
+    private static Channel newClient(EventLoopGroup group, SocketAddress address,
+            SslContext context, OcspClientCallback callback, ChannelHandler handler) {
+
+        Bootstrap bootstrap = new Bootstrap()
+                .channel(LocalChannel.class)
+                .group(group)
+                .handler(newClientHandler(context, callback, handler));
+
+        return bootstrap.connect(address)
+                .syncUninterruptibly()
+                .channel();
+    }
+
+    private static ChannelHandler newServerHandler(final SslContext context,
+            final byte[] response, final ChannelHandler handler) {
+        return new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) throws Exception {
+                ChannelPipeline pipeline = ch.pipeline();
+                SslHandler sslHandler = context.newHandler(ch.alloc());
+
+                if (response != null) {
+                    ReferenceCountedOpenSslEngine engine = (ReferenceCountedOpenSslEngine) sslHandler.engine();
+                    engine.setOcspResponse(response);
+                }
+
+                pipeline.addLast(sslHandler);
+
+                if (handler != null) {
+                    pipeline.addLast(handler);
+                }
+            }
+        };
+    }
+
+    private static ChannelHandler newClientHandler(final SslContext context,
+            final OcspClientCallback callback, final ChannelHandler handler) {
+        return new ChannelInitializer<Channel>() {
+            @Override
+            protected void initChannel(Channel ch) throws Exception {
+                ChannelPipeline pipeline = ch.pipeline();
+
+                SslHandler sslHandler = context.newHandler(ch.alloc());
+                ReferenceCountedOpenSslEngine engine = (ReferenceCountedOpenSslEngine) sslHandler.engine();
+
+                pipeline.addLast(sslHandler);
+                pipeline.addLast(new OcspClientCallbackHandler(engine, callback));
+
+                if (handler != null) {
+                    pipeline.addLast(handler);
+                }
+            }
+        };
+    }
+
+    private static boolean isCausedBy(Throwable t, Class<? extends Throwable> clazz) {
+        while (t != null) {
```

Comment:
consider using a `do {} while(...)` loop as it should not be called with a null `Throwable`